### Zadanie 1: przedział ufności dla $\mu_1-\mu_2$ przy znanych $\sigma_1, \sigma_2$

Gdy rozważamy dwie niezależne próby losowe $X_1, X_2, ..., X_{n_1}$ oraz $Y_1, Y_2, ..., Y_{n_2}$, które pochodzą odpowiednio z rozkładów $N(\mu_1,\sigma_1)$ oraz $N(\mu_2,\sigma_2)$, to statystyka zdefiniowana jako ${Z=\frac{(\bar{X}-\bar{Y})-(\mu_1-\mu_2)}{\sqrt{\frac{\sigma_1^2}{n_1}+\frac{\sigma_2^2}{n_2}}}}$ ma rozkład $N(0,1)$. Rozważając $\mathbb{P } (z_{\frac{\alpha}{2}} \leq Z \leq z_{1-\frac{\alpha}{2}})=1-\alpha$ i dokonując odpowiednich przekształceń otrzymamy przedział ufności dla różnicy średnich $\mu_1-\mu_2$ na poziomie ufności $1-\alpha$ ma postać $[(\bar{x}-\bar{y})-z_{1-\frac{\alpha}{2}}\sqrt{{\frac{\sigma_1^2}{n_1}+\frac{\sigma_2^2}{n_2}}},(\bar{x}-\bar{y})+z_{1-\frac{\alpha}{2}}\sqrt{{\frac{\sigma_1^2}{n_1}+\frac{\sigma_2^2}{n_2}}}]$

### Zadanie 2 a) (i), (iv), b) (i), (iv)

In [2]:
import numpy as np
import scipy.stats as stats

n1=50
n2=50
i_1=[0,1]
i_2=i_1
iv_1=i_1
iv_2=[1,2]
N=10000

def confidence_interval2(d1,d2, v1,v2, alpha=0.05):
    n1=len(d1)
    n2=len(d2)
    x=np.mean(d1)
    y=np.mean(d2)
    q_value=1-alpha/2
    quantile=stats.norm.ppf(q_value)
    s=np.sqrt(v1/n1+v2/n2)
    a=x-y
    return (a-quantile*s,a+quantile*s)

def task2(f,t1,t2,v1,v2):
    mi_in=0
    ci_len=[]
    for i in range(N):
        np.random.seed(i)
        d1=f(t1[0],t1[1],n1)
        d2=f(t2[0],t2[1],n2)
        a,b=confidence_interval2(d1,d2,v1,v2)
        mi=t1[0]-t2[0]
        if (mi<=b and mi>=a):
                mi_in+=1
        ci_len.append(b-a)
    print('Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności:', mi_in/N)
    print('Średnia długość przedziału ufności:', np.mean(ci_len))

print("a) - (i)")
task2(np.random.normal,i_1, i_2,i_1[1]**2,i_2[1]**2)
print("a) - (iv)")
task2(np.random.normal,iv_1, iv_2,iv_1[1]**2,iv_2[1]**2)


print("b) - (i)")
task2(np.random.logistic,i_1, i_2,(i_1[1]*np.pi)**2/3,(i_2[1]*np.pi)**2/3)
print("b) - (iv)")
task2(np.random.logistic,iv_1, iv_2,(iv_1[1]*np.pi)**2/3,(iv_2[1]*np.pi)**2/3)
    

a) - (i)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9495
Średnia długość przedziału ufności: 0.7839855938160216
a) - (iv)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9498
Średnia długość przedziału ufności: 1.2395900646091236
b) - (i)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9504
Średnia długość przedziału ufności: 1.4219925716322854
b) - (iv)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9525
Średnia długość przedziału ufności: 2.248367671099081


Możemy zaobserwować, że zarówno dla rozkładu normalnego, jak i logistycznego prawdopodobieństwo eksperymentalne jest bliskie teoretycznemu $1-\alpha$. Dla ustalonej liczności próby, długość przedziałów staje się większa, gdy rośnie wariancja któregoś z rozważanych rozkładów. Użycie kwantyli rozkładu normalego daje nam dostatecznie dobre wyniki również dla rozkładu logistycznego.

### Zadanie 5: przedział ufności dla $\mu_1-\mu_2$ przy nieznanych $\sigma_1, \sigma_2$

Dla znanych wariancji przedział ufności przyjął postać: $[(\bar{x}-\bar{y})-z_{1-\frac{\alpha}{2}}\sqrt{{\frac{\sigma_1^2}{n_1}+\frac{\sigma_2^2}{n_2}}},(\bar{x}-\bar{y})+z_{1-\frac{\alpha}{2}}\sqrt{{\frac{\sigma_1^2}{n_1}+\frac{\sigma_2^2}{n_2}}}]$. Teraz, gdy nie znamy wariancji przedział przyjmuje postać $[(\bar{x}-\bar{y})-t_{{1-\frac{\alpha}{2}},\nu}\sqrt{{\frac{s_1^2}{n_1}+\frac{s_2^2}{n_2}}},(\bar{x}-\bar{y})+t_{{1-\frac{\alpha}{2}},\nu}\sqrt{{\frac{s_1^2}{n_1}+\frac{s_2^2}{n_2}}}]$. Jak zwykle przy nieznanych wariancjach, używamy estymatorów wariancji $S_1^2$ i $S_2^2$. Przy wyliczaniu stopni swobody $\nu$ musimy uwzględnić wielkość prób i ich stopnie swobody. Liczba stopni swobody $\nu $ związana z tą estymatą wariancji jest przybliżana za pomocą równania Welcha-Satterthwaite’a: $\nu = \frac{\left( \frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}\right)^2}{\frac{s_1^4}{n_1^2 (n_1-1)}+\frac{s_2^4}{n_2^2 (n_2-1)}}$




### Zadanie 6

In [3]:
from scipy.stats import t

def confidence_interval5(d1,d2,alpha=0.05):
    n1=len(d1)
    n2=len(d2)
    x=np.mean(d1)
    y=np.mean(d2)
    q_value=1-alpha/2
    v1=np.var(d1, ddof=1)
    v2=np.var(d2, ddof=1)
    k=v1/n1+v2/n2
    s=np.sqrt(k)
    dof=k**2/(v1**2/(n1**2*(n1-1))+v2**2/(n2**2*(n2-1)))
    a=x-y
    quantile=t.ppf(q_value,dof)
    return (a-quantile*s,a+quantile*s)

def task5(f,t1,t2):
    mi_in=0
    ci_len=[]
    for i in range(N):
        np.random.seed(i)
        d1=f(t1[0],t1[1],n1)
        d2=f(t2[0],t2[1],n2)
        a,b=confidence_interval5(d1,d2)
        mi=t1[0]-t2[0]
        if (mi<=b and mi>=a):
                mi_in+=1
        ci_len.append(b-a)
    print('Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności:', mi_in/N)
    print('Średnia długość przedziału ufności:', np.mean(ci_len))

print("a) - (i)")
task5(np.random.normal,i_1, i_2)
print("a) - (iv)")
task5(np.random.normal,iv_1, iv_2)


print("b) - (i)")
task5(np.random.logistic,i_1, i_2)
print("b) - (iv)")
task5(np.random.logistic,iv_1, iv_2)
    

a) - (i)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9513
Średnia długość przedziału ufności: 0.7926278823248506
a) - (iv)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9508
Średnia długość przedziału ufności: 1.2576192564362176
b) - (i)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9507
Średnia długość przedziału ufności: 1.4351665676207086
b) - (iv)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9502
Średnia długość przedziału ufności: 2.2750073161305755


Możemy zaobserwować, że zarówno dla rozkładu normalnego, jak i logistycznego prawdopodobieństwo eksperymentalne jest bliskie teoretycznemu $1-\alpha$. Dostajemy wyniki względnie podobne do zadania 2, w którym wariancje były znane, zatem mimo braku znajomości wariancji, jesteśmy w stanie rozważać przedział ufności dla różnicy średnich z dużą dokładnością.

### Zadanie 9: przedział ufności dla $\frac{\sigma_2^2}{\sigma_1^2}$ przy nieznanych $\mu_1, \mu_2$ 

Rozważamy dwie niezależne próby losowe o rozkładach odpowiednio $N(\mu_1, \sigma_1)$ oraz $N(\mu_2, \sigma_2)$ o licznościach $n_1,n_2$. Do konstrukcji przedziału ufności dla ilorazu $\frac{\sigma_2^2}{\sigma_1^2}$ można użyć ilorazu $\frac{S_1^2}{S_2^2}$ dla odpowiednich wariancji próbkowych $S_1^2, S_2^2$. Wiemy, że wariancje próbkowe mają rozkład $\chi^2$. Skorzystamy z faktu, że iloraz zmiennych losowych $V_1,V_2$ rozkładzie $\chi^2$ ze stopniami swobody $\nu_1, \nu_2$, to zmienna losowa $\frac{V_1/\nu_1}{V_2/\nu_2}$ ma rozkład F (Snedecora) z $\nu_1$ i $\nu_2$ stopniami swobody. Wobec tego $Z=\frac{S_1^2/\sigma_1^2}{S_2^2/\sigma_2^2}$ ma rozkład $F$ z $\nu_1=n_1-1$ i $\nu_2=n_2-1$ stopniami swobody. Stąd   rozważając $\mathbb{P } (f_{{\frac{\alpha}{2},\nu_1,\nu_2}} \leq Z \leq f_{{1-\frac{\alpha}{2}},\nu_1,\nu_2})=1-\alpha$. Po przekształceniach otrzymujemy przedział ufności $[\frac{s_2^2}{s_1^2}f_{\alpha/2},\frac{s_2^2}{s_1^2}f_{1-\alpha/2} ]$.

### Zadanie 10

In [15]:
def confidence_interval10(d1,d2,alpha=0.05):
    n1=len(d1)
    n2=len(d2)
    q_1=alpha/2
    q_2=1-alpha/2
    v1=np.var(d1, ddof=1)
    v2=np.var(d2, ddof=1)
    s=v2/v1
    quantile1 = stats.f.ppf(q_1, n1-1, n2-1)
    quantile2 = stats.f.ppf(q_2, n1-1, n2-1)
    return (s*quantile1,s*quantile2)

def task10(f,t1,t2):
    si_in=0
    ci_len=[]
    for i in range(N):
        np.random.seed(i)
        d1=f(t1[0],t1[1],n1)
        d2=f(t2[0],t2[1],n2)
        a,b=confidence_interval10(d1,d2)
        si=(t2[1])**2/(t1[1])**2
        if (si<=b and si>=a):
                si_in+=1
        ci_len.append(b-a)
    print('Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności:', si_in/N)
    print('Średnia długość przedziału ufności:', np.mean(ci_len))

print("a) - (i)")
task10(np.random.normal,i_1, i_2)
print("a) - (iv)")
task10(np.random.normal,iv_1, iv_2)


print("b) - (i)")
task10(np.random.logistic,i_1, i_2)
print("b) - (iv)")
task10(np.random.logistic,iv_1, iv_2)
    

a) - (i)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9461
Średnia długość przedziału ufności: 0.0784523517003459
a) - (iv)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.9461
Średnia długość przedziału ufności: 0.3138094068013836
b) - (i)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.8741
Średnia długość przedziału ufności: 0.07842778150001772
b) - (iv)
Prawdopodobieństwo pokrycia ́sredniej przez przedział ufności: 0.8741
Średnia długość przedziału ufności: 0.3137111260000709


Możemy zaobserwować, że zarówno dla rozkładu normalnego eksperymentalne prawdopodobieństwo pokrycia jest bliskie teoretycznemu $1-\alpha$. W przypadku rozkładu logistycznego otrzymujemy $\approx 0.87$, więc mamy do czynienia z mniejszą dokładnością niż w poprzednich zadaniach. Może to wynikać z faktu, że czynimy założenia na temat normalności rozkładu zmiennej losowej, co prowadzi nas do użycia konkretnych rozkładów (chi-kwadrat), a potem ilorazu rozkładów chi-kwadrat (Snedecora), które niekoniecznie muszą odpowiadać rozkładom, które powstaną w wyniku rozważanych przez nas arytmetycznych przekształceń rozkładu logistycznego.

### Zadanie 12

Metoda delta dotyczy wyznaczania przybliżonego rozkładu prawdopodobieństwa funkcji asymptotycznie normalnego estymatora statystycznego na podstawie wiedzy o wariancji granicznej tego estymatora. Formalnie, twierdzenie  4.3.9 w podręczniku mówi, że dla ciągu zmiennych losowych ${X_n}$, który spełnia: $\sqrt{n}[X_n-\theta]\xrightarrow{D} N(0,\sigma^2)$ i funkcji różniczkowalnej w punkcie $\theta$ spełniającej $g'(\theta)\neq 0$. Wtedy zachodzi  $\sqrt{n}[g(X_n)-g(\theta)]\xrightarrow{D} N(0,\sigma^2[g'(\theta)]^2)$

**Przykład :**
Możemy zastosować tę metodę do wyznaczania przedziałów ufności - np. rozważmy zmienną losową $Z_n \sim \chi_n^2$. Wówczas unormowana zmienna losowa $Y=\frac{Z_n-n}{\sqrt{2n}}$ spełnia $\sqrt{n}{[\frac{1}{\sqrt{2}n}Z_n-\frac{1}{\sqrt{2}}]}\xrightarrow{D} N(0,1)$. Teraz, gdy chielibyśmy badać przedział ufności dla pierwiastka $Z_n$, możemy przyjąć $g(t)=\sqrt{t}$, $W_n=g(\frac{Z_n}{\sqrt{2}n})=(\frac{Z_n}{\sqrt{2}n})^{\frac{1}{2}}$, $g({\frac{1}{\sqrt{2}}})=\frac{1}{2^{1/4}}$, $g'({\frac{1}{\sqrt{2}}})=\frac{1}{2^{-3/4}}$. Zatem, stosując metodę delta, otrzymujemy: $\sqrt{n}[W_n-\frac{1}{2^{1/4}}]\xrightarrow{D} N(0,\frac{1}{2^{-3/4}})$, a to pozwoli nam zastosować kwantyle rozkładu normalnego o znanych już teraz parametrach przy wyliczaniu przedziału ufności. Graniczny rozkład normalny użyty przez nas w CTG będzie zadany właśnie tymi parametrami. 

**Podsumowując:** Możemy zastosować to twierdzenie do wyznaczania przedziałów ufności dla przekształconych zmiennych losowych, gdyż pozwala nam na zdeterminowanie paramtetrów kwantyli rozkładu normalnego, które definiują interesujące nas przedziały ufności.